.github/workflows/release.yml

```diff
@@ -0,0 +1,44 @@
+on:
+  label:
+    types: [created]
+
+# a lot of code taken from https://github.com/pypa/cibuildwheel/blob/main/examples/github-deploy.yml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+      - run: python -m pip install build
+      - run: python -m build
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: trio-sdist
+          path: dist/*.tar.gz
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: trio-wheel
+          path: dist/*.whl
```

Comment:
FWIW I don't think there's a case for putting the dists of a pure-python project into two separate GHA artifacts. Perhaps, you could combine them in a single step?